ui/app/pages/permissions-connect/choose-account/choose-account.component.js

```diff
@@ -76,31 +129,79 @@ export default class ChooseAccount extends Component {
     )
   }
 
-  render () {
-    const { originName, selectNewAccountViaModal, permissionsRequestId, cancelPermissionsRequest } = this.props
+  renderAccountsListHeader () {
     const { t } = this.context
+    const { selectNewAccountViaModal, accounts } = this.props
     return (
-      <div className="permissions-connect-choose-account">
-        <div className="permissions-connect-choose-account__title">
-          { t('chooseAnAcount') }
+      <div className="permissions-connect-choose-account__accounts-list-header">
+        <div className={classnames('permissions-connect-choose-account__select-all', {
+          invisible: accounts.length < 2,
```

Comment:
Nit: This is another case where we could avoid rendering instead of rendering invisibly.

Reply:
Done in f4faf0244